src/main/java/io/github/thebusybiscuit/slimefun4/core/networks/cargo/CargoNetworkTask.java

```diff
@@ -190,8 +192,12 @@ private ItemStack distributeItem(ItemStack stack, Location inputNode, List<Locat
             if (target.isPresent()) {
                 ItemStackWrapper wrapper = ItemStackWrapper.wrap(item);
                 item = CargoUtils.insert(network, inventories, output.getBlock(), target.get(), smartFill, item, wrapper);
+                index++;
 
                 if (item == null) {
+                    if (roundrobin) {
+                        network.roundRobin.put(inputNode, index % outputNodes.size());
```

Comment:
Why the modulo here?
Can you please elaborate and possibly add a comment to this part.

Reply:
The modulo is because the round robin index wraps around the output nodes, so once the index gets to the end of the output nodes it goes back to the start